conda_build/verify.py

```diff
@@ -0,0 +1,44 @@
+import os
+from subprocess import Popen, PIPE
+from .exceptions import VerifyError
+
+
+def get_exitcode_stdout_stderr(cmd):
+    proc = Popen(cmd, stdout=PIPE, stderr=PIPE)
+    out, err = proc.communicate()
+    exitcode = proc.returncode
+    return exitcode, out, err
+
+
+def is_valid_script(script):
+    if os.path.isfile(script) and script.split(".")[-1] == "py":
+        return True
+    return False
+
+
+def run_scripts(verify_scripts, verify_path, *args):
+    for script in verify_scripts:
+        script_path = os.path.join(verify_path, script)
+        print("Runinning script %s ..." % script)
+        cmd = ["python", script_path]
+        cmd.extend(list(args))
+        exitcode, out, err = get_exitcode_stdout_stderr(cmd)
+        if exitcode == 1:
+            raise VerifyError(str(err, 'utf-8'), script)
+
+
+def verify(verify_path, *args):
+    verify_scripts = [f for f in os.listdir(verify_path)
+                      if is_valid_script(os.path.join(verify_path, f))]
+    run_scripts(verify_scripts, verify_path, *args)
+    print("All scripts passed")
+
+
+def verify_package(path_to_package, config):
+    verify_path = os.path.join(config.verify_scripts_path, "package")
+    verify(verify_path, path_to_package)
+
+
+def verify_recipe(recipe, config):
```

Comment:
As we discussed, the thing that will probably actually be passed here is not the recipe itself, but rather the rendered metadata.  That means that instead of doing a subprocess to run each script, it would be better to lay out some standard function that each verification script must implement.  Then, import each verify script and call that particular function, passing in the metadata with it.

For an example of how to do imports dynamically, see https://github.com/conda/conda-build/blob/master/conda_build/api.py#L144-L182

The alternative is to write the rendered yaml to a file, and then run the verification in subprocesses like you do here.  It's not as nice.